controllers/operator_controller.go

```diff
@@ -47,7 +51,59 @@ type OperatorReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.1/pkg/reconcile
 func (r *OperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	l := log.FromContext(ctx).WithName("reconcile")
+	l.V(1).Info("starting")
+	defer l.V(1).Info("ending")
+
+	var existingOp = &operatorsv1alpha1.Operator{}
+	if err := r.Get(ctx, req.NamespacedName, existingOp); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	reconciledOp := existingOp.DeepCopy()
+	res, reconcileErr := r.reconcile(ctx, reconciledOp)
+
+	// Do checks before any Update()s, as Update() may modify the resource structure!
+	updateStatus := !equality.Semantic.DeepEqual(existingOp.Status, reconciledOp.Status)
+	updateFinalizers := !equality.Semantic.DeepEqual(existingOp.Finalizers, reconciledOp.Finalizers)
+
+	// Compare resources - ignoring status & metadata.finalizers
+	compareOp := reconciledOp.DeepCopy()
+	existingOp.Status, compareOp.Status = operatorsv1alpha1.OperatorStatus{}, operatorsv1alpha1.OperatorStatus{}
+	existingOp.Finalizers, compareOp.Finalizers = []string{}, []string{}
+	specDiffers := !equality.Semantic.DeepEqual(existingOp, compareOp)
```

Comment:
Is compareOp needed? Could you not compare the specs directly?
```suggestion
	specDiffers := !equality.Semantic.DeepEqual(&existingOp.Spec, &reconciled.Spec)
```

Reply:
We're not comparing just specs, but specs and the metadata as well. Want to make sure no annotations or labels or etc were added.

Reply:
If we didn't care about the metadata, then that would be correct.

Reply:
This assumes that we want to allow changes to the metadata field.

Reply:
In this case, it's _only_ the finalizers that can change. Any other metadata change will cause a `panic()`

Reply:
`specDiffers` could be better named.

Reply:
How about `unexpectedFieldsChanged`?

Reply:
Unresolved only because I'm not sure you'd see my comment. Not merge blocking though.

Reply:
I see it!